stdlib/test/collections/test_optional.mojo

```diff
@@ -151,10 +151,12 @@ def test_optional_equality():
     assert_true(o == 10)
     assert_true(o != 11)
     assert_true(o != n)
-    assert_true(o != None)
+    # TODO: Find a way to allow "o != None" again.
+    assert_true(o.__ne__(None))
```

Comment:
in the meantime we can recommend `is None` which works well.

Reply:
I think this will be fixed when we change the compiler to emit `NoneType()` when it sees `None`. (Currently it emits the `#kgen.none` raw MLIR type, which means there's always one extra "hop" of conversion where `NoneType` is currently used.)